internal_transfers/actions/src/orderbook.ts

```diff
@@ -0,0 +1,69 @@
+import { WinningSettlementData } from "./models";
+import axios from "axios";
+
+// https://api.cow.fi/docs/#/
+const PROD_API_URL =
+  "https://api.cow.fi/mainnet/api/v1/solver_competition/by_tx_hash";
+const BARN_API_URL =
+  "https://barn.api.cow.fi/mainnet/api/v1/solver_competition/by_tx_hash";
+
+async function getWinningSettlementFromOrderbookAPI(
+  apiUrl: string
+): Promise<WinningSettlementData | undefined> {
+  try {
+    const response = await axios.get(apiUrl);
+
+    // The Orderbook, stores all solution submissions sorted by the objective criteria.
+    // The winning solution is the last entry of the `solutions` array.
+    const solutionArray = response.data.solutions;
+    const winningSolution = solutionArray[solutionArray.length - 1];
```

Comment:
What if the solutionArray is 0?

Reply:
Very good question -- Probably better for @sunce86 (backend-solver folks) but I certainly hope that is not the case. Anyway, this table `solver_competitions` will have to be productionized for this service to work. 

I am taking a quick peek now to see if this has even happened before with the following query:

```
select *
from solver_competitions
where jsonb_array_length(json -> 'solutions') = 0
```

I think its going to be a while.

Reply:
I will put an assertion here to catch the issue, but really hope it never happens. We will need to keep track of assertions being made through the workflow so we know exactly how it can fail.